examples/main.cpp

```diff
@@ -0,0 +1,222 @@
+   const int NE = mesh.GetNE();
+   const int ndofs = x_ref.Size() / NE;
+
+   MassIntegrator mass_int;
+   Vector VecMassMats(ndofs * ndofs * NE);
+   mass_int.AssembleEA(fespace, VecMassMats, false);
+
+   //Batch solver...
+   DenseTensor LUMassMats(ndofs, ndofs, NE);
+   DenseTensor MassMats(ndofs, ndofs, NE);
+   Array<int> P(rhs.Size());
+
+   for (int i=0; i<ndofs * ndofs * NE; ++i)
+   {
+      LUMassMats.HostWrite()[i] = VecMassMats.HostRead()[i];
+      MassMats.HostWrite()[i] = VecMassMats.HostRead()[i];
+   }
+
+   //
+   //Compute reference solution
+   //
+   mfem::BatchLUFactor(LUMassMats, P);
+   mfem::BatchLUSolve(LUMassMats, P, x_ref);
+
+
+   //
+   //Compute Magma solution
+   //
+
+   //Compute magma version with variable batch solver
+   magma_int_t magma_device = 0;
+   magma_queue_t magma_queue;
+
+   magma_setdevice(magma_device);
+   magma_queue_create(magma_device, &magma_queue);
+
+
+   //Number of rows and columns of each matrix
+   Array<magma_int_t> num_of_rows(NE);
+   Array<magma_int_t> num_of_cols(NE);
+
+   //Pointers to mass matrices
+   Array<double *> magma_LUMassMats(NE);
+   for (int i=0; i<NE; ++i)
+   {
+      num_of_rows[i] = ndofs;
+      num_of_cols[i] = ndofs;
+      magma_LUMassMats.HostWrite()[i] = &MassMats.ReadWrite()[i*ndofs*ndofs];
+   }
+
+   Array<magma_int_t> ldda = num_of_cols;
+   Array<magma_int_t *> dipiv_array(NE);
+   Array<magma_int_t> info_array(NE);
+
+   //Pointers to pivot vectors
+   P.HostReadWrite();
+   P = 0.0;
+   for (int i=0; i<NE; ++i)
+   {
+      dipiv_array.HostWrite()[i] = &P.ReadWrite()[i*ndofs];
+   }
+
+   //Perform variable batch size factorization
+   magma_dgetrf_vbatched(num_of_rows.ReadWrite(), num_of_cols.ReadWrite(),
+                         magma_LUMassMats.ReadWrite(), ldda.ReadWrite(),
+                         dipiv_array.ReadWrite(), info_array.ReadWrite(),
+                         NE, magma_queue);
+
+
+   //
+   //Question: How do we perform the batch LU/Solve ?
+   //
+
+
+   //rhs only has 1 column
+   Array<int> rhs_num_of_cols(NE);
+   rhs_num_of_cols = 1;
+
+   //Question: Do we need to perform the pivoting ourselfs?
+
+   //x_magma_ptrs to x_magma
+   Array<double *> x_magma_ptrs(NE);
+   for (int i=0; i<NE; ++i)
+   {
+      x_magma_ptrs.HostWrite()[i] = &x_magma.ReadWrite()[i*ndofs];
+   }
+
+   Array<magma_int_t> trSolve_ldda(NE + 1);
+   Array<magma_int_t> trSolve_lddb(NE + 1);
+   for (int i=0; i<NE+1; ++i)
+   {
+      trSolve_ldda[i] = ndofs; //??
+      trSolve_lddb[i] = ndofs; //??
+   }
+
+   //
+   //Current error:
+   // hip error code: 'hipErrorInvalidValue':1 at /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/rocBLAS/library/src/rocblas_auxiliary.cpp:633
+   //
+
+
+   //L(Ux) = b //lower solve
+   magmablas_dtrsm_vbatched(MagmaLeft, MagmaLower, MagmaNoTrans,
+                            MagmaNonUnit,
+                            num_of_rows.ReadWrite(),
+                            rhs_num_of_cols.ReadWrite(),
```

Comment:
I believe that error is coming from `magma_getvector`, which is called in `magmablas_dtrsm_vbatched`: 

```
    // compute the max. dimensions
    magma_imax_size_2(m, n, batchCount, queue);
    magma_int_t max_m, max_n;
    magma_getvector(1, sizeof(magma_int_t), &m[batchCount], 1, &max_m, 1, queue);
    magma_getvector(1, sizeof(magma_int_t), &n[batchCount], 1, &max_n, 1, queue);
    ```
    so both the `m` and `n` arrays need to be of size `NE + 1`.  The `magma_imax_size_2` function will put the max value in the "extra" spot at the end.  That's why `trsm_vbatched` is trying to copy the last elements of the arrays, assuming that the arrays are of length `batchCount + 1`.
    
  So, both `num_of_rows` and `rhs_num_of_cols` need to be 1 element larger, but you don't have to initialize the last element (MAGMA will do that).